### PR TITLE
Setup-Associations PR

### PR DESCRIPTION
**Pull request by Javed-Shahbaz**
## In this milestone, 
- Finish creating the remaining classes for our school library.
- Set up associations between classes and objects

### CODE REVIEW

File: rent.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  attr_accessor :date, :person, :books
+
+  def initialize(date, person, books)
+    @date = date
+    @person = person
+    @book = books
+    books.rentals << self
+    person.rentals << self
+  end
+end
```
Comment (alexander16108) [inline rent.rb:11]: a method for creating rentals must be present in the person.rb file and the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association`👍🏾

File: books.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def ad_rental(rental)
+    @rentals.push(rental)
+    rental.book = self
+  end
+end
```
Comment (alexander16108) [inline books.rb]: To create a rental here we need the person and date and the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association

```
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: classroom.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  def initialize(label)
+    @label = label
+    @students = []
+  end
+
+  def ad_student(student)
+    students << student
+    student.classroom = self
+  end
+end
```
Comment (alexander16108) [inline classroom.rb:14]: Add a method that serves as the 'belongs-to' side (a student belongs to a classroom) to the student.rb file 👍🏾

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @Javed-Shahbaz  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Followed git workflow
- No linter errors
- Created the remaining classes

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codecaiine — APPROVED** [review]
Hi @Javed-Shahbaz , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good associations setup ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**